includes/embeds/class-amp-youtube-embed-handler.php

```diff
@@ -82,117 +94,313 @@ public function unregister_embed() {
 	 *
 	 * @param string $cache Cache for oEmbed.
 	 * @param string $url   Embed URL.
+	 *
 	 * @return string Embed.
 	 */
 	public function filter_embed_oembed_html( $cache, $url ) {
-		$id = $this->get_video_id_from_url( $url );
-		if ( ! $id ) {
-			return $cache;
+
+		if ( empty( $cache ) || empty( $url ) ) {
+			return '';
 		}
 
-		$props = $this->parse_props( $cache, $url, $id );
-		if ( empty( $props ) ) {
+		$video_id = $this->get_video_id_from_url( $url );
+
+		if ( ! $video_id ) {
 			return $cache;
 		}
 
-		$props['video_id'] = $id;
-		return $this->render( $props, $url );
+		return $this->render( $cache, $url );
 	}
 
 	/**
-	 * Parse AMP component from iframe.
+	 * Convert YouTube iframe into AMP YouTube component.
 	 *
-	 * @param string $html     HTML.
-	 * @param string $url      Embed URL, for fallback purposes.
-	 * @param string $video_id YouTube video ID.
-	 * @return array|null Props for rendering the component, or null if unable to parse.
+	 * @param string $html HTML markup of YouTube iframe.
+	 * @param string $url  YouTube URL.
+	 *
+	 * @return string HTML markup of AMP YouTube component.
 	 */
-	private function parse_props( $html, $url, $video_id ) {
-		$props = $this->match_element_attributes( $html, 'iframe', [ 'title', 'height', 'width' ] );
-		if ( ! isset( $props ) ) {
-			return null;
+	public function render( $html, $url ) {
+
+		$attributes = $this->prepare_attributes( $url );
+
+		$iframe_props = [ 'title', 'height', 'width' ];
+		$props        = $this->match_element_attributes( $html, 'iframe', $iframe_props );
+		foreach ( $iframe_props as $iframe_prop ) {
+			if ( ! empty( $props[ $iframe_prop ] ) ) {
+				$attributes[ $iframe_prop ] = $props[ $iframe_prop ];
+			}
 		}
```

Comment:
This could be moved to the `prepare_attributes()` method as it seems more relevant there.

Reply:
These three attributes parse differently in `render()` (from HTML markup) and in `get_amp_component()` (from DOMElement). That is a reason it's parse saperatlly.